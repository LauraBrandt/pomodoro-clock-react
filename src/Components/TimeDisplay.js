import React from 'react';

const TimeDisplay = ({type, time}) => {
  return(
    <div>
      <label htmlFor="time-left" id="timer-label">{type}</label>
      <div id="time-left">{time}</div>
    </div>
  );
}

export default TimeDisplay;