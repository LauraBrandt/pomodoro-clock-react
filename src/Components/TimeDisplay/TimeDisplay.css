.time-display {
  border: 5px solid #9e0000;
  border-radius: 50%;
  width: 350px;
  height: 350px;
  margin: 0 auto 1em auto;
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  overflow: hidden;
}

.time-display-fill {
  position: absolute;
  bottom: 0;
  width: 100%;
  height: 100%;
  background: #ba0000;
  z-index: -1;
  transition: height 1s linear;
}

#timer-label {
  font-size: 2rem;
  text-transform: uppercase;
}

#time-left {
  font-size: 5rem;
  padding: 1rem 0;
}

.last-ten-seconds {
  color: #ba0000;
}

@media (max-width: 450px) {
  .time-display {
    width: 300px;
    height: 300px;
    margin: 1.5em auto 1em auto;
  }
}