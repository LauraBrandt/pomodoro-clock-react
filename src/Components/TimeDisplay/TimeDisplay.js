import React from 'react';
import PropTypes from 'prop-types';
import './TimeDisplay.css';

export function formatTime(seconds) {
  const minutes = Math.floor(seconds/60);
  const secondsRemaining = seconds % 60;
  return `${minutes}:${secondsRemaining < 10 ? '0' : ''}${secondsRemaining}`;
}

const TimeDisplay = ({type, time, percentHeight}) => {
  return(
    <div className='time-display'>
      <div className='time-display-fill' style={{height: `${percentHeight}%`}}></div>
      <label htmlFor="time-left" id="timer-label">{type}</label>
      <div id="time-left">{formatTime(time)}</div>
    </div>
  );
}

TimeDisplay.propTypes = {
  type: PropTypes.string.isRequired,
  time: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  percentHeight: PropTypes.number
}

export default TimeDisplay;