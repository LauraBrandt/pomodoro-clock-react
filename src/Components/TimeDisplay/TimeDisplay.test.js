import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import TimeDisplay, {formatTime} from './TimeDisplay';

it('renders without crashing', () => {
  shallow(<TimeDisplay type="" time=""/>);
});

describe('structure', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = mount(<TimeDisplay type="Session" time={25*60}/>);
  });
  
  it('has all elements with ids necessary', () => {
    expect(wrapper.find('#timer-label')).toHaveLength(1);
    expect(wrapper.find('#time-left')).toHaveLength(1);
  });
  
  it('sets the label correctly', () => {
    expect(wrapper.find('#timer-label').text()).toEqual('Session');
  });

  it('sets the time correctly', () => {
    expect(wrapper.find('#time-left').text()).toEqual("25:00");
  });

  it('has the correct time format (mm:ss)', () => {
    const time = wrapper.find('#time-left').text();
    expect(time).toMatch(/\d+:\d{2}/)
  });
});

describe('formatTime', () => {
  it('calculates and formats the display time correctly', () => {
    const timeString = formatTime(1000);
    expect(timeString).toBe('16:40');
  });

  it('zero-pads seconds', () => {
    const timeString = formatTime(608);
    expect(timeString).toMatch(/\d+:\d{2}/);
    expect(timeString).toBe('10:08');
  });

  it("doesn't zero-pad minutes", () => {
    const timeString = formatTime(500);
    expect(timeString).toMatch(/\d{1}:\d{2}/);
    expect(timeString).toBe('8:20');
  });
});

describe('', () => {
  it('adds class to turn red in last 10 seconds', () => {
    const wrapper = shallow(<TimeDisplay type="Session" time={10}/>);
    expect(wrapper.find('.last-ten-seconds')).toHaveLength(1);
    wrapper.setProps({ time: 0 });
    expect(wrapper.find('.last-ten-seconds')).toHaveLength(1);
  });

  it('removes last-ten-seconds class when time > 10', () => {
    const wrapper = shallow(<TimeDisplay type="Session" time={11}/>);
    expect(wrapper.find('.last-ten-seconds')).toHaveLength(0);
  });
});

describe('snapshot', () => {
  it('renders', () => {
    const component = renderer.create(<TimeDisplay type="Session" time={488}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
