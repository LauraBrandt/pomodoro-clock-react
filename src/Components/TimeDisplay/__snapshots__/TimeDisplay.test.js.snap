// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders 1`] = `
<div
  className="time-display"
>
  <div
    className="time-display-fill"
    style={
      Object {
        "height": "undefined%",
      }
    }
  />
  <label
    htmlFor="time-left"
    id="timer-label"
  >
    Session
  </label>
  <div
    className=""
    id="time-left"
  >
    8:08
  </div>
</div>
`;
